tests/e2e/tests/plugins/JavaPlugin.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { TestConstants } from '../../TestConstants';
+import { ProjectTree } from '../../pageobjects/ide/ProjectTree';
+import * as projectAndFileTests from '../../testsLibrary/ProjectAndFileTests';
+import { Key } from 'selenium-webdriver';
+import { Editor } from '../../pageobjects/ide/Editor';
+
+const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
+const ide: Ide = e2eContainer.get(CLASSES.Ide);
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree);
+const editor: Editor = e2eContainer.get(CLASSES.Editor);
+
+const devfileUrl: string = 'https://gist.githubusercontent.com/Ohrimenko1988/4432cb57b084512e681bd27fddcf6b74/raw/8fb1e6db3df7a09d4b96a449dd12a2977168bd3c/java11PluginTest.yaml';
+const factoryUrl: string = `${TestConstants.TS_SELENIUM_BASE_URL}/f?url=${devfileUrl}`;
+const codeNavigationClassName: string = 'String.class';
+const sampleName: string = 'console-java-simple';
+const subRootFolder: string = 'src';
+
+const fileFolderPath: string = `${sampleName}/src/main/java/org/eclipse/che/examples`;
+const tabTitle: string = 'HelloWorld.java';
+
+suite(`The 'JavaPlugin' test`, async () => {
+    suite('Create workspace', async () => {
+        test('Create workspace using factory', async () => {
+            await driverHelper.navigateToUrl(factoryUrl);
+        });
+
+        test('Wait until created workspace is started', async () => {
+            await ide.waitAndSwitchToIdeFrame();
+            await ide.waitIde(TimeoutConstants.TS_SELENIUM_START_WORKSPACE_TIMEOUT);
+            await projectTree.openProjectTreeContainer();
+            await projectTree.waitProjectImported(sampleName, subRootFolder);
+        });
+    });
+
+    suite('Test opening file', async () => {
+        // opening file that soon should give time for LS to initialize
+        projectAndFileTests.openFile(fileFolderPath, tabTitle);
+    });
+
+    suite('Language server validation', async () => {
+        test('Wait until Java LS is initialised', async () => {
+            await ide.checkLsInitializationStart('Activating');
+            await ide.waitStatusBarTextAbsence('Activating', 900_000);
+        });
+
+        test('Autocomplete', async () => {
+            await editor.moveCursorToLineAndChar(tabTitle, 10, 20);
+            await editor.pressControlSpaceCombination(tabTitle);
+            await editor.waitSuggestionContainer();
+            await editor.waitSuggestion(tabTitle, 'close() : void');
+        });
+
+        test('Error highlighting', async () => {
+            const textForErrorDisplaying: string = '$';
+            await editor.type(tabTitle, textForErrorDisplaying, 10);
+            await editor.waitErrorInLine(10, TimeoutConstants.TS_ERROR_HIGHLIGHTING_TIMEOUT);
+            await editor.performKeyCombination(tabTitle, Key.chord(Key.BACK_SPACE));
+            await editor.waitErrorInLineDisappearance(10);
+        });
+
+        test('Codenavigation', async () => {
```

Comment:
Actually the test is verifying "**Suggestion invoking**" not "**Codenovigation**".

Reply:
According to the description of solving issue https://github.com/eclipse/che/issues/19437 the test should check both "Code navigation" and "Suggestion invoking:.

Reply:
"Ctrl + F12" is a "Open definition" function it opens class definition. Is that not code navigation?
Deleted "Suggeston invoking" from the issue

Reply:
You are right. "_Ctrl+f12_" stands for "**Go to Implementations**". 
It wasn't clear for me that test verified movement to String.class implementation.

Reply:
"Suggestion invoking" is still in the issue description:
![Screenshot from 2021-03-30 16-28-58](https://user-images.githubusercontent.com/1197777/112997495-d70bad80-9175-11eb-87af-ab5f758b3e75.png)

Reply:
Done.